File: 2026POTD/JANUARY/09-01-2026/865.cs
public class Solution {
    private (int depth, TreeNode node) Dfs(TreeNode root) {
        if (root == null) return (0, null);
        var l = Dfs(root.left);
        var r = Dfs(root.right);
        if (l.depth > r.depth) return (l.depth + 1, l.node);
        if (r.depth > l.depth) return (r.depth + 1, r.node);
        return (l.depth + 1, root);
    }

    public TreeNode SubtreeWithAllDeepest(TreeNode root) {
        return Dfs(root).node;
    }
}

File: 2026POTD/JANUARY/09-01-2026/865.js
var subtreeWithAllDeepest = function(root) {
    function dfs(node) {
        if (!node) return [0, null];
        const [ld, ln] = dfs(node.left);
        const [rd, rn] = dfs(node.right);
        if (ld > rd) return [ld + 1, ln];
        if (rd > ld) return [rd + 1, rn];
        return [ld + 1, node];
    }
    return dfs(root)[1];
};